File: platform/core-api/src/com/intellij/openapi/application/Result.java
// Copyright 2000-2021 JetBrains s.r.o. and contributors. Use of this source code is governed by the Apache 2.0 license.
package com.intellij.openapi.application;

import com.intellij.util.ThrowableRunnable;

/**
 * @deprecated Use {@link WriteAction#run(ThrowableRunnable)} or {@link ReadAction#run(ThrowableRunnable)} or similar method instead
 */
@Deprecated
public class Result<T> {

  protected T myResult;

  public final void setResult(T result) {
    myResult = result;
  }

}

File: fleet/util/core/srcCommonMain/fleet/util/Result.kt
// Copyright 2000-2024 JetBrains s.r.o. and contributors. Use of this source code is governed by the Apache 2.0 license.
package fleet.util


inline fun <T, R> Result<T>.andThen(next: (T) -> Result<R>): Result<R> = fold(
  onSuccess = { next(it) },
  onFailure = { Result.failure(it) }
)

inline fun<T> Result<T>.wrapError(next: (Throwable) -> Throwable): Result<T> = fold(
  onSuccess = { Result.success(it) },
  onFailure = { Result.failure(next(it)) }
)